code/build/template/include/extensions/[+python orxPy.h]

```diff
@@ -0,0 +1,2613 @@
+    else                                      \
+    {                                         \
+      PY_CHECK_ARG_TYPE(index, tp_orxobject); \
+      ARG_PTR(orxOBJECT *, p##name, index);   \
+      name = *p##name;                        \
+    }                                         \
+  } while (0)
+#define ARG_CLOCK(name, index)             \
+  orxCLOCK *name;                          \
+  do                                       \
+  {                                        \
+    PY_CHECK_ARG_TYPE(index, tp_orxclock); \
+    ARG_PTR(orxCLOCK *, p##name, index);   \
+    name = *p##name;                       \
+  } while (0)
+
+// Binding return value handling
+#define RETURN_VALUE(CONV, RET) \
+  do                            \
+  {                             \
+    CONV(py_retval(), RET);     \
+    return true;                \
+  } while (0)
+#define RETURN_STR(RET) RETURN_VALUE(py_newstr, RET)
+#define RETURN_FLOAT(RET) RETURN_VALUE(py_newfloat, RET)
+#define RETURN_INT(RET) RETURN_VALUE(py_newint, RET)
+#define RETURN_BOOL(RET) RETURN_VALUE(py_newbool, RET)
+#define RETURN_NONE          \
+  do                         \
+  {                          \
+    py_newnone(py_retval()); \
+    return true;             \
+  } while (0)
+#define RETURN_OBJECT_OR_NONE(OBJ)       \
+  do                                     \
+  {                                      \
+    if (OBJ == orxNULL)                  \
+    {                                    \
+      RETURN_NONE;                       \
+    }                                    \
+    else                                 \
+    {                                    \
+      py_neworxobject(py_retval(), OBJ); \
+      return true;                       \
+    }                                    \
+  } while (0)
+#define RETURN_VECTOR(VEC)             \
+  do                                   \
+  {                                    \
+    py_neworxvector(py_retval(), VEC); \
+    return true;                       \
+  } while (0)
+#define RETURN_CLOCK(CLOCK)             \
+  do                                    \
+  {                                     \
+    py_neworxclock(py_retval(), CLOCK); \
+    return true;                        \
+  } while (0)
+
+  // Function wrappers
+
+  // Helpers
+
+  void py_neworxobject(py_OutRef pyOut, orxOBJECT *_pstObject)
+  {
+    orxOBJECT **ppstObject = (orxOBJECT **)py_newobject(pyOut, tp_orxobject, 0, sizeof(orxOBJECT *));
+    *ppstObject = _pstObject;
+  }
+
+  void py_neworxvector(py_OutRef pyOut, orxVECTOR *_pvVector)
+  {
+    orxVECTOR *pvVector = (orxVECTOR *)py_newobject(pyOut, tp_orxvector, 0, sizeof(orxVECTOR));
+    orxVector_Copy(pvVector, _pvVector);
+  }
+
+  void py_neworxclock(py_OutRef pyOut, orxCLOCK *_pstClock)
+  {
+    orxCLOCK **ppstClock = (orxCLOCK **)py_newobject(pyOut, tp_orxclock, 0, sizeof(orxCLOCK *));
+    *ppstClock = _pstClock;
+  }
+
+  // Object functions

```

Comment:
The following functions are defining each bound module's functions. The `orxObject_*` bindings make up most of these definitions.